cranelift/codegen/src/machinst/abi.rs

```diff
@@ -12,6 +12,12 @@ pub trait ABIBody {
     /// The instruction type for the ISA associated with this ABI.
     type I: VCodeInst;
 
+    /// How many temps are needed?
+    fn needed_tmps(&self) -> usize;
```

Comment:
This might have been made with future-proofing in mind, but in this case, since we only need 0 or 1 temporary all the time, it seems that this could just return a boolean right now, and `init_with_tmps` could take a single `tmp` argument. Alternatively, could we pass a closure function to `init_with_tmps` that returns a tmp, so that it can claim as many as it requires? (and then `needed_tmps` wouldn't be needed at all)

(Implementing any of these ideas would avoid the short-lived SmallVec allocation below.)